### DISCUSSION TITLE
How can I run the OpenROAD GUI on Ubuntu locally and not in Docker?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#2679] [General]
How can I install and run the OpenROAD GUI locally?

I want the advantages of Docker, not having to install everything, easy to deploy upgrades, but when I run the OpenROAD GUI from docker, the experience is pretty good, but not quite as good as running locally on Ubuntu.

Another problem is that the OpenROAD-flow-scripts sets up a TON of environment variables, even if I had OpenROAD GUI installed locally, how could I get the envioronment set up correctly as I build various designs?

Example of how I run the OpenROAD GUI in Docker. To run my own designs, I map results, log, objects, designs to my local drive by modifying the script in the github issue below.

https://github.com/The-OpenROAD-Project/OpenROAD/issues/2675#issue-1513733033

Ideally I'd like a debian package for the OpenROAD GUI. It seems like it should be possible to host a debian package on github, but it is a bit of work, especially considering that it probably needs to be built nightly to be truely useful in the open soruce development process: https://assafmo.github.io/2019/05/02/ppa-repo-hosted-on-github.html

**Comment by QuantamHD** [comment]
Creating a snap package in the CI might be a good option https://ubuntu.com/tutorials/create-your-first-snap#1-overview

**Reply by oharboe** [reply]
snap packages are specific to Ubuntu, as far as I know. debian packages might have broader reach. Ideally, CI would create multiple package formats (rpm and deb?) for broadest possible reach.

**Reply by QuantamHD** [reply]
You can install snap on almost any distro, but whatever you think best address the need. 

I thought snap would be a good option because we could update it frequently.

**Comment by maliberty** [comment]
You would run the [dependency installer](https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/etc/DependencyInstaller.sh) and then compiler OR locally  (build_openroad.sh --local).

I'm surprised using docker makes much of a difference for the GUI.  What do you see?

You can always mount your directory in docker so envars shouldn't be a big issue.

**Comment by mithro** [comment]
Have you tried the conda packages?

**Reply by mithro** [reply]
https://github.com/hdl/conda-eda

**Reply by mithro** [reply]
FYI - @proppy

**Reply by proppy** [reply]
conda-eda packages currently don't include the GUI.

**Reply by proppy** [reply]
https://github.com/hdl/conda-eda/pull/228
